### ISSUE TITLE
Consider making equals & toHash (and possibly toString) `const` in hunt.lang.Object

### ISSUE THREAD

**Issue by n8sh** [huntlabs/hunt#34]
That the standard D `Object.equals(Object)` and `Object.toHash()` methods aren't `const` is a historical source of problems. The Hunt framework has an opportunity to correct this in `IObject` & `AbstractObject`. `toString` is another method that might make sense to force to be `const`.

**Comment by zoujiaqing** [comment]
update to hunt 1.0.0-rc.8

**Comment by Heromyth** [comment]
Yes, we have tried to do what you suggest, and give up finally. The constraints from const or immutable are too strong, and the implements in compiler and phobos are still weak. For example, we have to including the whole std.array for the reason of const, see: https://github.com/huntlabs/hunt/blob/master/source/hunt/container/array.d. 

More discuss: 
http://jmdavisprog.com/articles/why-const-sucks.html
https://forum.dlang.org/post/mailman.581.1520247475.3374.digitalmars-d-announce@puremagic.com

So, it's not a good time to using const.
